frontend/src/lib-components/molecules/FileUpload.tsx

```diff
@@ -303,6 +318,20 @@ export default React.memo(function FileUpload({
     files.map(attachmentToFile)
   )
 
+  useEffect(() => {
+    if (!onStateChange) return
+
+    onStateChange({
+      success: uploadedFiles.filter(
+        (f) => f.uploaded && !f.error && !f.deleteInProgress
+      ).length,
+      inProgress: uploadedFiles.filter(
+        (f) => (!f.uploaded && !f.error) || f.deleteInProgress
+      ).length,
+      error: uploadedFiles.filter((f) => f.error).length
+    })
+  }, [onStateChange, uploadedFiles])
```

Comment:
Tämä olis parempi tehdä siellä missä kutsutaan `setUploadedFiles`:iä, ks. esim: https://react.dev/learn/you-might-not-need-an-effect#notifying-parent-components-about-state-changes